README.md

```diff
@@ -39,30 +39,118 @@ extern crate rss;
 
 ## Reading
 
-A channel can be read from any object that implements the `BufRead` trait. 
+### From a `BufRead`
+
+A channel can be read from any object that implements the `BufRead` trait.
 
 ```rust
-let reader: BufRead = ...;
+use std::fs::File;
+use std::io::BufReader;
+use rss::Channel;
+
+let file = File::open("tests/data/rss2sample.xml").unwrap();
+let reader = BufReader::new(file);
 let channel = Channel::read_from(reader).unwrap();
 ```
 
+### From a URL
+
+A channel can also be read from a URL.
+
+To enable this functionality you must enable the `from_url` feature in your Cargo.toml.
+
+```toml
+[dependencies]
+rss = { version = "*", features = ["from_url"] }
+```
+
+```ignore
+use rss::Channel;
+
+let channel = Channel::from_url("https://feedpress.me/usererror.xml").unwrap();
+```
+
 ## Writing
 
 A channel can be written to any object that implements the `Write` trait or converted to an XML string using the `ToString` trait.
 
 **Note**: Writing a channel does not perform any escaping of XML entities.
 
+### Example
+
 ```rust
-let channel: Channel = ...;
-let writer: Write = ...;
+use std::fs::File;
+use std::io::{BufReader, sink};
+use rss::Channel;
 
-// write the channel to a writer
-channel.write_to(writer).unwrap();
+let file = File::open("tests/data/rss2sample.xml").unwrap();
+let reader = BufReader::new(file);
+let channel = Channel::read_from(reader).unwrap();
+
+// write to the channel to a writer
+channel.write_to(sink()).unwrap();
 
 // convert the channel to a string
 let string = channel.to_string();
 ```
 
+## Creation
+
+A channel can be created using the Builder functions.
+
+### Example
+
+```
+use rss::{ChannelBuilder, ImageBuilder};
+
+let image = ImageBuilder::default()
+    .url("http://jupiterbroadcasting.com/images/LAS-300-Badge.jpg")
+    .title("LAS 300 Logo")
+    .link("http://www.jupiterbroadcasting.com")
+    .finalize();
+
+let channel = ChannelBuilder::default()
+    .title("The Linux Action Show! OGG")
+    .link("http://www.jupiterbroadcasting.com")
+    .description("Ogg Vorbis audio versions of The Linux Action Show!")
+    .image(image)
+    .finalize();
+```
+
+## Validation
+
+Validation can be performed using either a `Channel` or a builder.
+
+The the following checks are performed during validation:
+
+* Ensures that integer properties can be parsed from their string representation into integers
+* Ensures that the integer properties are within their valid range according to the RSS 2.0 specification
+* Ensures that URL properties can be parsed
+* Ensures that string properties where only certain values are allowed fall within those valid values
+
+### Example
+
+```
+use rss::Channel;
+
+let input = include_str!("tests/data/rss2sample.xml");
+let channel = input.parse::<Channel>().unwrap();
+channel.validate().unwrap();
+```
+
+### Example
```

Comment:
Instead of having two `Example` sections back to back, you could do something like this if you wanted:

```
### Examples

Validating a `<channel>`:

'''
<code here>
'''

Validating an `<image>`:

'''
<code here>
'''
```